### PR TITLE
Revised python script and automated update of SUMMARY.MD

### LINKED ISSUE #11: Enhance generate-api-docs.py to automatically update SUMMARY.MD
Modify the generate-api-docs.py script to automatically update the appropriate sections of SUMMARY.MD.

### PR DESCRIPTION
**Pull request by dsbeach**
Closes #11 

display_functions renamed lcd functions

updated script to replace relevent sections of LUADOC

updated SUMMARY.md as a result of revised script (input was opentx/master)

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -26,37 +26,56 @@
...
-       * [getValue()](general/getValue.md)
-       * [getVersion()](general/getVersion.md)
-       * [killEvents()](general/killEvents.md)
-       * [playDuration()](general/playDuration.md)
-       * [playFile()](general/playFile.md)
-       * [playNumber()](general/playNumber.md)
-       * [playTone()](general/playTone.md)
-       * [popupInput()](general/popupInput.md)
+
+[//]: <> (LUADOC-BEGIN:general)
```
Comment (projectkk2glider) [inline SUMMARY.md]: Wow, how did you know that this would not show in gitbook?

File: tools/generate-api-docs.py [hunk 2]
```diff
@@ -16,6 +16,10 @@
 MODULES = {}
 FUNCTIONS = []
 
+STARTMARKER = "[//]: <> (LUADOC-BEGIN:"
+ENDMARKER = "[//]: <> (LUADOC-END:"
+SUMMARYFILE = "SUMMARY.MD"
```
Comment (projectkk2glider) [inline tools/generate-api-docs.py]: Little problem on linux, which is case sensitive. Please change to "SUMMARY.md"

File: tools/generate-api-docs.py [hunk 3]
```diff
@@ -202,6 +206,40 @@ def mkdir_p(path):
...
+def insertSection(newContents, sectionName):
+  print("Inserting section: %s" % sectionName)
+  newContents.append("%s%s)\n\n" % (STARTMARKER, sectionName))
+  newContents.append("   * [%s Functions](%s/%s_functions.md)\n" % (sectionName.capitalize(), sectionName, sectionName))
+
+  for f in sorted(MODULES[sectionName]):
+    # f = (moduleName, funcName, funcDefinition, description, params, retvals, notices)
+    newContents.append("      * [%s](%s/%s.md)\n" % (f[2].rstrip(), sectionName, f[1]))
+    foo = "      * [%s](%s/%s.md)\n" % (f[2].rstrip(), sectionName, f[1])
+    print(foo)
```
Comment (projectkk2glider) [inline tools/generate-api-docs.py]: Use logDebug instead of print

File: SUMMARY.md (hunk 1 above)
Comment (dsbeach) [inline SUMMARY.md]: See http://stackoverflow.com/questions/4823468/comments-in-markdown

**Comment by projectkk2glider** [comment]
Ok, very good. I will merge it now.

**Comment by projectkk2glider** [comment]
@dsbeach one more bug found! Every time a script is run every section in SUMMARY.md gets a new empty line added. Can you fix that before the merge?

**Comment by projectkk2glider** [comment]
Also add `Closes #11` into your opening post. This will then automatically close the original issue when the merge is done and also link it with this one.

**Comment by dsbeach** [comment]
Hopefully we are good to go now.

Python is a pretty interesting language for that sort of thing.  I'm sure it is a whole lot easier to maintain than some of the perl code I hacked together before I retired.  The Python support in Visual Studio was nice too.

**Comment by projectkk2glider** [comment]
Yes Python is great, you can do a lot with a small amount of code. The pull is perfect now.

**Pull request merged**